### ISSUE TITLE
Allow HTTPClient to intercept multiple redirects

### ISSUE THREAD

**Issue by kylebrowning** [swift-server/async-http-client#604]
Title says it all. Tesla's Auth API requires 2 redirects to be followed, one of which has a token in as a query parameter in the URL.

It seems with the current implementation of HTTPClient I can only stop all redirects, and follow them myself.

It would be nice to just have a history available of all redirects following when a request is finished.

**Comment by Lukasa** [comment]
Yeah, fine-grained control over the redirect process is something I've wanted for a long time. Building this feature out fully is a little complex, as we'd need support for pluggable redirect strategies and some way to pass data around, as well as simply a history. A history, however, is something we can definitely provide.

In the meantime, is following the redirects yourself sufficient?

**Comment by gaintap** [comment]
@kylebrowning  Would be curious how you're following the redirects. I'm not seeing the URL in the response with async-http like you would see with URLSession. Do you have to get it off of a delegate?
